sql/schema.go

```diff
@@ -141,27 +153,71 @@ func (s *Schema) Migrate(logger *zap.Logger, db Database, before, vacuumState in
 					return fmt.Errorf("apply %s: %w", m.Name(), err)
 				}
 			}
-			// version is set intentionally even if actual migration was skipped
-			if _, err := tx.Exec(fmt.Sprintf("PRAGMA user_version = %d;", m.Order()), nil, nil); err != nil {
-				return fmt.Errorf("update user_version to %d: %w", m.Order(), err)
+			if err := s.setVersion(tx, m.Order()); err != nil {
+				return err
 			}
 			return nil
 		}); err != nil {
-			err = errors.Join(err, db.Close())
-			return err
+			return errors.Join(err, db.Close())
```

Comment:
The only caller of this method will always call `db.Close` even when an error happens, so I don't think we need to do it here and join the errors 🙂

Reply:
Fixed